kube-derive/tests/crd_mixed_enum_test.rs

```diff
@@ -0,0 +1,396 @@
+#[derive(CustomResource, Serialize, Deserialize, Debug, Clone, JsonSchema)]
+#[kube(group = "clux.dev", version = "v1", kind = "InvalidEnum5")]
+#[serde(untagged)]
+enum InvalidEnum5Spec {
+    A { inner: String },
+    B { inner: u32 },
+}
+
+/// This enum is valid, as the `inner` fields are the same.
+#[derive(CustomResource, Serialize, Deserialize, Debug, Clone, JsonSchema)]
+#[kube(group = "clux.dev", version = "v1", kind = "ValidEnum6")]
+#[serde(untagged)]
+enum ValidEnum6Spec {
+    A {
+        /// The inner fields need to have the same schema (so also same description)
+        inner: u32,
+        /// An additional field
+        additional: String,
+    },
+    B {
+        /// The inner fields need to have the same schema (so also same description)
+        inner: u32,
+    },
+}
+
+/// This enum is invalid, as the typed of `inner` fields are the same, *but* the description (which
+/// is part of the schema differs).
+#[derive(CustomResource, Serialize, Deserialize, Debug, Clone, JsonSchema)]
+#[kube(group = "clux.dev", version = "v1", kind = "InvalidEnum7")]
+#[serde(untagged)]
+enum InvalidEnum7Spec {
+    A {
+        /// The inner fields need to have the same schema (so also same description)
+        inner: u32,
+        additional: String,
+    },
+    B {
+        /// This description differs!
+        inner: u32,
+    },
+}
+
+#[tokio::test]
+#[ignore = "needs apiserver to validate CRDs"]
+async fn check_are_valid_crds() {
+    let crds = [ValidEnum3::crd(), ValidEnum4::crd(), ValidEnum6::crd()];
+
+    let client = Client::try_default()
+        .await
+        .expect("failed to create Kubernetes client");
+    let crd_api: Api<CustomResourceDefinition> = Api::all(client);
+    for crd in crds {
+        // Clean up existing CRDs. As these are only test CRDs and this test is not run by default
+        // this is fine.
+        let _ = crd_api.delete(&crd.name_any(), &DeleteParams::default()).await;
+
+        // Prevent "object is being deleted: customresourcedefinition already exists
+        tokio::time::sleep(Duration::from_millis(100)).await;
+
+        crd_api
+            .create(&PostParams::default(), &crd)
+            .await
+            .expect("failed to create CRD");
+    }
+}
+
+/// Use `cargo test --package kube-derive print_crds -- --nocapture` to get the CRDs as YAML.
+/// Afterwards you can use `kubectl apply -f -` to see if they are valid CRDs.
+#[test]
+fn print_crds() {
+    println!("{}", serde_yaml::to_string(&ValidEnum3::crd()).unwrap());
+    println!("---");
+    println!("{}", serde_yaml::to_string(&ValidEnum4::crd()).unwrap());
+}
+
+#[test]
+#[should_panic = "oneOf variants must all have the same type"]
+fn invalid_enum_1() {
+    InvalidEnum1::crd();
+}
+
+#[test]
+#[should_panic = "oneOf variants must all have the same type"]
+fn invalid_enum_2() {
+    InvalidEnum2::crd();
+}
+
+#[test]
+fn valid_enum_3() {
+    assert_json_eq!(
+        ValidEnum3::crd(),
+        json!(
```

Comment:
you might be able to make these integration tests a little less intimidating by only testing the main version output;


```suggestion
        ValidEnum3::crd().spec.versions[0],
        json!( ...skip first 16 lines
```